dotnet/src/SemanticKernel/AI/OpenAI/Clients/OpenAIClientAbstract.cs

```diff
@@ -35,15 +35,17 @@ public abstract class OpenAIClientAbstract : IDisposable
     protected HttpClient HTTPClient { get; }
 
     private readonly HttpClientHandler _httpClientHandler;
-    private readonly IHttpRetryPolicy _retryPolicy = new NullRetryPolicy();
+    private readonly DelegatingHandler _retryHandler = new DefaultHttpRetryHandler();
 
-    internal OpenAIClientAbstract(ILogger? log = null, IHttpRetryPolicy? retryPolicy = null)
+    internal OpenAIClientAbstract(ILogger? log = null, DelegatingHandler? retryHandler = null)
     {
         this.Log = log ?? this.Log;
-        this._retryPolicy = retryPolicy ?? this._retryPolicy;
+        this._retryHandler = retryHandler ?? this._retryHandler;
 
         this._httpClientHandler = new() { CheckCertificateRevocationList = true };
-        this.HTTPClient = new HttpClient(this._httpClientHandler);
+        this._retryHandler.InnerHandler = this._httpClientHandler;
```

Comment:
Found a bug, DelegatingHandler cannot be shared, so changing to implement a Factory so each HTTPClient can create its own.

Reply:
Fixed.